pkg/daemon/daemon.go

```diff
@@ -187,6 +185,7 @@ func New(
 		exitCh:                 exitCh,
 		stopCh:                 stopCh,
 	}
+	dn.atomicSSHKeysWriter = dn.atomicallyWriteSSHKey
```

Comment:
Could you explain what this is for?

Reply:
this is a lightweight version of the filesystem client, as in, the client existed mainly to be able to unit test the flow as we shouldn't wrap the file management API of Golang. By having a function which deals with writes only, we're able to mock it out in unit test till we have proper e2e in place for these scenarios.

Reply:
ah thanks! I missed the note on line 87.